internal/mesh/internal/types/proxy_configuration.go

```diff
@@ -15,5 +17,35 @@ func RegisterProxyConfiguration(r resource.Registry) {
 		Scope: resource.ScopeNamespace,
 		// TODO(rb): add validation for proxy configuration
 		Validate: nil,
+		Mutate:   MutateProxyConfiguration,
 	})
 }
+
+func MutateProxyConfiguration(res *pbresource.Resource) error {
+	var proxyCfg pbmesh.ProxyConfiguration
+	err := res.Data.UnmarshalTo(&proxyCfg)
+	if err != nil {
+		return resource.NewErrDataParse(&proxyCfg, err)
+	}
+
+	changed := false
+
+	// Default the tproxy outbound port.
+	if proxyCfg.IsTransparentProxy() {
```

Comment:
This was previously defaulted in the sidecar proxy controller code instead of the mutating webhook. Moving it here as that is a more appropriate place to apply defaults like this.

Reply:
Doesn't this mean that if you do not write a `ProxyConfiguration` then you get different defaults than if you have an empty `ProxyConfiguration` targeting a workload?

Reply:
I don't think so. If you have an empty one, then tproxy is still false and so this mutation would not take place.